### PR TITLE
Handling for cross-package imports of "export name".

### PR DESCRIPTION
**Pull request by jonmeow**
Starts warning on duplicate "export name" because I'm updating name lookup to make it work.

Need to think harder about how to handle cross-package "export import". But adding some failing tests to sketch out what *should* work.

### CODE REVIEW

File: toolchain/diagnostics/diagnostic_consumer.cpp [hunk 1]
```diff
@@ -19,8 +19,15 @@ auto StreamDiagnosticConsumer::HandleDiagnostic(Diagnostic diagnostic) -> void {
...
-      *stream_ << "ERROR: ";
+    switch (message.level) {
+      case DiagnosticLevel::Error:
+        *stream_ << "ERROR: ";
+        break;
+      case DiagnosticLevel::Warning:
+        *stream_ << "WARNING: ";
+        break;
+      case DiagnosticLevel::Note:
+        break;
```
Comment (chandlerc) [inline toolchain/diagnostics/diagnostic_consumer.cpp]: Could you split this out to its own PR? Mostly because it causess some churn in diagnostics and it seems a bit cleaner to have those separate from the import part.

Comment (jonmeow) [reply toolchain/diagnostics/diagnostic_consumer.cpp]: Done, #3972

**Comment by jonmeow** [comment]
Now depends on #3972 [ed: now merged]

File: toolchain/check/testdata/packages/no_prelude/cross_package_export.carbon [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+// CHECK:STDERR: ^~~~~~
+// CHECK:STDERR: export_name_copy.carbon:4:1: In import.
+// CHECK:STDERR: import library "base";
+// CHECK:STDERR: ^~~~~~
+// CHECK:STDERR: base.carbon:4:1: ERROR: Duplicate name being declared in the same scope.
+// CHECK:STDERR: class C {
+// CHECK:STDERR: ^~~~~~~~~
+// CHECK:STDERR: fail_todo_use_export_name_both.carbon:[[@LINE-10]]:1: In import.
+// CHECK:STDERR: import Other library "export_name";
+// CHECK:STDERR: ^~~~~~
```
Comment (chandlerc) [inline toolchain/check/testdata/packages/no_prelude/cross_package_export.carbon:100]: Not about this PR at all, but maybe for future stuff -- it's a bit awkward taht we end up doing the import walk twice here. Is it once for the error and once for the note? If so, I at least have an idea of how this might be addressed.

**Review by chandlerc — APPROVED** [review]
Nice, LG!

(The comment below is totally unrelated to this PR)
